darwin/exporter/formats/yolo_segmented.py

```diff
@@ -0,0 +1,236 @@
+                continue
+
+            annotation_type = YoloSegmentedAnnotationType.POLYGON
+        else:
+            annotation_type = YoloSegmentedAnnotationType.UNKNOWN
+
+        Point = namedtuple("Point", ["x", "y"])
+
+        if annotation.data is None:
+            logger.warn(f"Skipped annotation at index {annotation_index} because it's data fields are empty.'")
+            continue
+
+        data = annotation.data
+        points: List[Point] = []
+
+        if annotation_type == YoloSegmentedAnnotationType.BOUNDING_BOX:
+            logger.debug(f"Exporting bounding box at index {annotation_index}.")
+
+            try:
+                # Create 8 coordinates for the x,y pairs of the 4 corners
+                x1, y1, x2, y2, x3, y3, x4, y4, x5, y5 = (
+                    data["x"],
+                    data["y"],
+                    (data["x"] + data["w"]),
+                    (data["y"] + data["h"]),
+                    (data["x"] + data["w"]),
+                    data["y"],
+                    data["x"],
+                    (data["y"] + data["h"]),
+                    data["x"],
+                    data["y"],
+                )
+
+                logger.debug(
+                    "Coordinates for bounding box: "
+                    f"({x1}, {y1}), ({x2}, {y2}), "
+                    f"({x3}, {y3}), ({x4}, {y4}), "
+                    f"({x5}, {y5})"  # Unsure if we have to close this.
+                )
+
+                # Normalize the coordinates to a proportion of the image size
+                n_x1 = normalise(x1, im_w)
+                n_y1 = normalise(y1, im_h)
+                n_x2 = normalise(x2, im_w)
+                n_y2 = normalise(y2, im_h)
+                n_x3 = normalise(x3, im_w)
+                n_y3 = normalise(y3, im_h)
+                n_x4 = normalise(x4, im_w)
+                n_y4 = normalise(y4, im_h)
+                n_x5 = normalise(x5, im_w)
+                n_y5 = normalise(y5, im_w)
+
+                logger.debug(
+                    "Normalized coordinates for bounding box: "
+                    f"({n_x1}, {n_y1}), ({n_x2}, {n_y2}), "
+                    f"({n_x3}, {n_y3}), ({n_x4}, {n_y4}), "
+                    f"({n_x5}, {n_y5})"
+                )
+
+                # Add the coordinates to the points list
+                points.append(Point(x=n_x1, y=n_y1))
+                points.append(Point(x=n_x2, y=n_y2))
+                points.append(Point(x=n_x3, y=n_y3))
+                points.append(Point(x=n_x4, y=n_y4))
+
+                if CLOSE_VERTICES:
+                    points.append(Point(x=n_x5, y=n_y5))
+
+            except KeyError as exc:
+                logger.warn(
+                    f"Skipped annotation at index {annotation_index} because an"
+                    "expected key was not found in the data.",
+                    exc_info=exc,
+                )
```

Comment:
also would extract to helper to improve readability

Reply:
Yeah, originally I did, but time was against me on this one